crates/sui-core/src/execution_engine.rs

```diff
@@ -200,6 +201,11 @@ fn transfer<S>(
     recipient: SuiAddress,
 ) -> SuiResult {
     object.transfer(recipient)?;
+    temporary_store.log_event(SuiEvent::TransferCoin {
```

Comment:
Should this generalized to `SuiEvent::TransferObject`? We will definitely want an event like that as well (since "tell me what objects my address received" is a key use-case of the event service, and I don't think it makes sense to have separate events for coins and objects (unless the coin one has coin-specific metadata like amount and coin type).

Reply:
Ok, I unified all transfers into a single TransferObject, with a field to distinguish by type of transfer (Coin, ToAddress, ToObject for wraps)